nixos/modules/installer/cd-dvd/sd-image.nix

```diff
@@ -130,27 +122,25 @@ in
 
   config = {
     fileSystems = {
-      "/boot/firmware" = {
-        device = "/dev/disk/by-label/${config.sdImage.firmwarePartitionName}";
+      "/boot" = {
+        device = "/dev/disk/by-label/${config.sdImage.bootPartitionName}";
```

Comment:
There is a reason it is `/boot/firmware`, it is because it is an "opaque partition" that is used exclusively to hold u-boot for the raspberry pi. This partition **has** to be removed for Rockchip boards as their bootloader installed to an arbitrary location encroaches on that partition. This would break usage on Rockchip boards.

Additionally, having a separate boot partition is painful, unless we enlarge it enough to hold enough generations. (Which I see you have done).

Though, really it would be preferable to not have `/boot` as another partition by default.